src/blocks/blocks.service.ts

```diff
@@ -117,6 +118,50 @@ export class BlocksService {
     return { ...block, transactions };
   }
 
+  async batchUpdateGrafitti(input: BatchUpdateGraffitiDto): Promise<Block[]> {
+    const networkVersion = this.config.get<number>('NETWORK_VERSION');
+
+    const blocks = await this.prisma.block.findMany({
+      where: {
+        hash: {
+          in: input.updates.map((update) => standardizeHash(update.hash)),
+        },
+        network_version: networkVersion,
+      },
+    });
+
+    if (blocks.length !== input.updates.length) {
+      throw new NotFoundException();
+    }
+
+    const hashGraffitiMap = new Map<string, string>();
+    input.updates.forEach((update) => {
+      hashGraffitiMap.set(update.hash, update.graffiti);
+    });
+
+    blocks.map((block) => {
+      const graffiti = hashGraffitiMap.get(block.hash);
+      if (graffiti) {
+        block.graffiti = graffiti;
+      }
+    });
+
+    await this.prisma.$transaction([
```

Comment:
One thing I'd be careful about here is putting too many updates into a single transactions. Transactions that are too big can really stall out the database just FYI